optimize/backend/src/main/java/io/camunda/optimize/service/db/os/report/interpreter/groupby/process/identity/AbstractProcessGroupByIdentityInterpreterOS.java

```diff
@@ -0,0 +1,192 @@
+import io.camunda.optimize.service.db.os.report.filter.util.ModelElementFilterQueryUtilOS;
+import io.camunda.optimize.service.db.os.report.interpreter.RawResult;
+import io.camunda.optimize.service.db.os.report.interpreter.groupby.process.AbstractProcessGroupByInterpreterOS;
+import io.camunda.optimize.service.db.report.ExecutionContext;
+import io.camunda.optimize.service.db.report.interpreter.groupby.process.identity.ProcessGroupByIdentityInterpreterHelper;
+import io.camunda.optimize.service.db.report.plan.process.ProcessExecutionPlan;
+import io.camunda.optimize.service.db.report.result.CompositeCommandResult;
+import io.camunda.optimize.service.db.report.result.CompositeCommandResult.DistributedByResult;
+import io.camunda.optimize.service.db.report.result.CompositeCommandResult.GroupByResult;
+import io.camunda.optimize.service.util.configuration.ConfigurationService;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.opensearch.client.opensearch._types.FieldValue;
+import org.opensearch.client.opensearch._types.aggregations.Aggregate;
+import org.opensearch.client.opensearch._types.aggregations.Aggregation;
+import org.opensearch.client.opensearch._types.aggregations.FilterAggregate;
+import org.opensearch.client.opensearch._types.aggregations.NestedAggregate;
+import org.opensearch.client.opensearch._types.aggregations.StringTermsAggregate;
+import org.opensearch.client.opensearch._types.aggregations.StringTermsBucket;
+import org.opensearch.client.opensearch._types.aggregations.TermsAggregation;
+import org.opensearch.client.opensearch._types.query_dsl.Query;
+import org.opensearch.client.opensearch.core.SearchResponse;
+
+public abstract class AbstractProcessGroupByIdentityInterpreterOS
+    extends AbstractProcessGroupByInterpreterOS {
+  private static final String GROUP_BY_IDENTITY_TERMS_AGGREGATION = "identities";
+  private static final String FLOW_NODES_AGGREGATION = "flowNodeInstances";
+  private static final String USER_TASKS_AGGREGATION = "userTasks";
+  private static final String FILTERED_USER_TASKS_AGGREGATION = "filteredUserTasks";
+  // temporary GROUP_BY_IDENTITY_MISSING_KEY to ensure no overlap between this label and userTask
+  // names
+  private static final String GROUP_BY_IDENTITY_MISSING_KEY = "unassignedUserTasks___";
+
+  protected abstract ConfigurationService getConfigurationService();
+
+  protected abstract DefinitionService getDefinitionService();
+
+  protected abstract ProcessGroupByIdentityInterpreterHelper getHelper();
+
+  @Override
+  public Map<String, Aggregation> createAggregation(
+      final Query query,
+      final ExecutionContext<ProcessReportDataDto, ProcessExecutionPlan> context) {
+    final TermsAggregation termsAggregation =
+        new TermsAggregation.Builder()
+            .size(
+                getConfigurationService()
+                    .getElasticSearchConfiguration()
```

Comment:
This should be getOpenSearchConfoguration() configuration. This is present in C7 as well, please just make sure it gets corrected with your next PR there as well